File: src/main/java/com/koddy/server/global/aop/DailyMailAuthLimit.java
package com.koddy.server.global.aop;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import java.util.concurrent.TimeUnit;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface DailyMailAuthLimit {
    int maxTry() default 3;

    int banTime() default 30;

    TimeUnit banTimeUnit() default TimeUnit.MINUTES;
}

File: src/main/java/com/koddy/server/global/aop/DailyMailAuthLimit.kt
package com.koddy.server.global.aop

import java.util.concurrent.TimeUnit

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class DailyMailAuthLimit(
    val maxTry: Int = 3,
    val banTime: Long = 10,
    val banTimeUnit: TimeUnit = TimeUnit.MINUTES,
)
